### PR TITLE
adjust to TSC verbosity API changes

### PR DESCRIPTION
**Pull request by tomerd**
motivation: deprecate global and process verbosity

changes:
* stop using TSC global verbosity flags
* set a logging handler on Process to funnel debug information to observability handler
* adjust call sites

### CODE REVIEW

**Comment by tomerd** [comment]
goes with https://github.com/apple/swift-tools-support-core/pull/290

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please smoke test

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please smoke test

File: Tests/CommandsTests/BuildToolTests.swift [hunk 1]
```diff
@@ -345,19 +345,19 @@ final class BuildToolTests: CommandsTestCase {
...
 
             // Now try building using XCBuild while specifying a faulty compiler override.  This should fail.  Note that we need to set the executable to use for the manifest itself to the default one, since it defaults to SWIFT_EXEC if not provided.
             var overriddenOutput = ""
             do {
-                overriddenOutput = try execute(["-c", "debug", "-v"], environment: ["SWIFT_EXEC": "/usr/bin/false", "SWIFT_EXEC_MANIFEST": ToolchainConfiguration.default.swiftCompilerPath.pathString], packagePath: path).stdout
+                overriddenOutput = try execute(["-c", "debug", "--vv"], environment: ["SWIFT_EXEC": "/usr/bin/false", "SWIFT_EXEC_MANIFEST": ToolchainConfiguration.default.swiftCompilerPath.pathString], packagePath: path).stdout
                 XCTFail("unexpected success (was SWIFT_EXEC not overridden properly?)")
             }
-            catch SwiftPMProductError.executionFailure(let error, let stdout, _) {
+            catch SwiftPMProductError.executionFailure(let error, _, let stderr) {
```
Comment (tomerd) [inline Tests/CommandsTests/BuildToolTests.swift:357]: this is now going to stderr as its a "proper" debug diagnostics

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please smoke test

File: Sources/Commands/SwiftTool.swift [hunk 2]
```diff
@@ -433,17 +433,8 @@ public class SwiftTool {
...
-        case .debug:
-            TSCUtility.verbosity = .debug
-        case .info:
-            TSCUtility.verbosity = .verbose
-        case .warning, .error:
-            TSCUtility.verbosity = .concise
-        }
-        Process.verbose = TSCUtility.verbosity != .concise
+        // set global process logging handler
+        Process.loggingHandler = { self.observabilityScope.emit(debug: $0) }
```
Comment (tomerd) [inline Sources/Commands/SwiftTool.swift:437]: this means the process arguments will only be logged in _debug_ mode or iow require --vv to be shown

**Comment by tomerd** [comment]
I guess self-hosted do not work with cross-repo tests, so expected to fail

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please test

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please smoke test

File: Sources/Commands/SwiftTool.swift (hunk 2 above)
Comment (abertelrud) [reply Sources/Commands/SwiftTool.swift:465]: I think this is good — it's been annoying that _all_ process invocations such as `vtool` to discover SDK info or manifest compilation gets logged with `-v`.

Comment (abertelrud) [reply Sources/Commands/SwiftTool.swift:465]: Probably something to note in the release notes.

**Review by abertelrud — APPROVED** [review]

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please smoke test

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/290

@swift-ci please smoke test

**Comment by tomerd** [comment]
> I guess self-hosted do not work with cross-repo tests, so expected to fail

@shahmishal is it possible to have the self-hosted checks work with cross-repo?

**Pull request merged**